server/hotrod/src/main/java/org/infinispan/server/hotrod/iteration/DefaultIterationManager.java

```diff
@@ -87,17 +96,33 @@ public void segmentCompleted(Set<Integer> segments) {
       this.metadata = metadata;
       this.resultFunction = resultFunction;
    }
+
+   @Override
+   public void close() {
+      stream.close();
+   }
 }
 
 
 public class DefaultIterationManager implements IterationManager {
 
    private static final Log log = LogFactory.getLog(DefaultIterationManager.class, Log.class);
 
-   private final Map<String, IterationState> iterationStateMap = new ConcurrentHashMap<>();
+   private final com.github.benmanes.caffeine.cache.Cache<String, IterationState> iterationStateMap;
    private final Map<String, KeyValueFilterConverterFactory> filterConverterFactoryMap =
          new ConcurrentHashMap<>();
 
+   public DefaultIterationManager(TimeService timeService) {
+      Caffeine<Object, Object> builder = Caffeine.newBuilder();
+      builder.expireAfterAccess(1, TimeUnit.MINUTES).removalListener(new RemovalListener<Object, Object>() {
```

Comment:
I wonder if we should increase to like 5 minutes? 1 minute seems a bit low for a default you can't change. Although we should probably make this configurable at some point.